lib/google/auth/extras/impersonated_credential.rb

```diff
@@ -44,36 +75,75 @@ def initialize(email_address:, scope:, base_credentials: nil, delegate_email_add
             transform_email_to_name(email)
           end
 
-          @impersonate_lifetime = lifetime
+          if token_type == :id_token
+            @impersonate_include_email = include_email
+          elsif !include_email.nil?
+            raise ArgumentError, 'Can only provide include_email when using target_audience'
```

Comment:
FWIU, `token_type == :id_token` means we are using target_audience. Does this and below error need to be rephrased? Or swapped.

```
      # The token type as symbol, either :id_token or :access_token
      def token_type
        target_audience ? :id_token : :access_token
      end
```

Reply:
Ah, sorry i think i get this now. this might help?

```
          if token_type == :id_token # using target audience
            @impersonate_include_email = include_email
          elsif !include_email.nil? # not using target audience and include_email is provided.
            raise ArgumentError, 'Can only provide include_email when using target_audience'
          end
```

Reply:
And for below

```ruby
          if token_type == :access_token # not using target audience
            @impersonate_lifetime = lifetime
          elsif !lifetime.nil? # using target audience and lifetime is provided
            raise ArgumentError, 'Cannot provide lifetime when using target_audience'
          end
```